WcaOnRails/app/models/user.rb

```diff
@@ -508,6 +508,11 @@ def editable_fields_of_user(user)
       fields << :senior_delegate_id
       fields << :region
     end
+    if user.any_kind_of_delegate? && (user == self || user.senior_delegate == self || admin? || board_member)
+      fields << :location_description
+      fields << :phone_number
+      fields << :notes
```

Comment:
I've just updated the code with another idea. Please, let me know whether this looks ok or we should go for something else =)